presto-prometheus/src/main/java/io/prestosql/plugin/prometheus/PrometheusSplitManager.java

```diff
@@ -153,4 +215,77 @@ static long timeUnitsToMillis(String timeWithUnits)
     {
         return Duration.valueOf(timeWithUnits).toMillis();
     }
+
+    private static class EffectiveLimits
+    {
+        private ZonedDateTime now;
+        private String maxQueryRangeDurationStr;
+        private Optional<PrometheusPredicateTimeInfo> maybePredicateRange;
+        private ZonedDateTime nowOrUpperBound;
```

Comment:
I can't help but feel like we should just call this `relativeUpperBound` and change `now` to `absoluteUpperBound` or something agnostic of the idea that it's "now". and rename `getNowOrUpperBound` to `getUpperBound` which returns the calculated option.

Reply:
Yep, agreed. I thought about it a bit and `defaultUpperBound` made the most sense to me. Changed in: 220d624 . And all the comments and method named reflect the refactoring. Comments note that `defaultUpperBound `might be `now()` if nothing else sets it.